shared/mail/mail.go

```diff
@@ -291,6 +293,11 @@ func SendMail(c smtpClient, mail mailData, date time.Time) error {
 	htmlMessage := mail.htmlBody
 
 	txtBody, err := html2text.FromString(mail.htmlBody)
+
+	// could be replaced by a call to GetSiteURL() but I have no idea how to do that...
```

Comment:
The `config.HostName` in `*SMTPConfig` contains the hostname of the SiteURL. We can pass that when we call this function.

This also means we need to change the function signature of an exported function. But since we are passing unexported types, there's actually no point in keeping this exported. Let's unexport this, and pass the smtpConfig

Reply:
Thanks for your suggestion, but I think that exceeds my Go coding skills, so I won't be able to provide an updated patch for this suggestion, sorry.

Reply:
No worries, thank you for your time @anx-ag !

Reply:
Also I didn't see https://github.com/mattermost/mattermost-server/pull/18232, so I think this one can be closed, since the other one is more sophisticated anyways.

Reply:
#18232 only adds message ids for some emails, not all. We need to cover all the code paths.